File: server/python/app/presentation/models/payment_link/response.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from pydantic.alias_generators import to_camel

class PaymentLinkResponse(BaseModel):
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    payment_link_id: str
    redirect_url: str
    status: Optional[str] = None
    amount: Optional[float] = None
    currency: Optional[str] = None

File: server/ruby/app/presentation/models/payment_link/response.rb
module Presentation
  module Models
    module PaymentLink
      class Response
        attr_accessor :payment_link_id, :redirect_url, :status, :amount, :currency

        def initialize(payment_link_id: nil, redirect_url: nil, status: nil, amount: nil, currency: nil)
          @payment_link_id = payment_link_id
          @redirect_url    = redirect_url
          @status          = status
          @amount          = amount
          @currency        = currency
        end

        def as_json(*)
          {
            payment_link_id: @payment_link_id,
            redirect_url:    @redirect_url,
            status:          @status&.to_s,
            amount:          @amount,
            currency:        @currency&.to_s
          }
        end

        def to_json(*options)
          as_json.to_json(*options)
        end
      end
    end
  end
end